### PR TITLE
Feat: Allow csv and json input files fix#6

### LINKED ISSUE #6: Allow other input file types
Currently we are only allowed to use `.txt` input files.
It would be nice to allow other types, such as `csv` or `json`.
Then, it would be required a new setting to set the field/column to be loaded as URL.
For this, we should follow the same idea as `SPIDERFEEDER_FILE_HANDLERS` to allow the users to easily extend the usage to new file types.

### PR DESCRIPTION
**Pull request by ejulio**
Fix #6 
This PR allows input files in CSV and JSON formats.
To load them properly it is required to set `SPIDERFEEDER_INPUT_FIELD` setting.
Also, the user can extend the built-in parsers through `SPIDERFEEDER_FILE_PARSERS`.

### CODE REVIEW

File: tests/test_loaders.py [hunk 1]
```diff
@@ -130,3 +139,52 @@ def test_file_encoding(get_crawler, mocker):
...
+        'SPIDERFEEDER_INPUT_FILE': 'local.myext',
+        'SPIDERFEEDER_FILE_PARSERS': {
+            'myext': 'tests.test_loaders.CustomParser'
+        }
+    })
+    StartUrlsLoader.from_crawler(crawler)
+
+    crawler.signals.send_catch_log(signals.spider_opened, spider=crawler.spider)
+
+    assert crawler.spider.start_urls == ['http://override.com']
```
Comment (ejulio) [inline tests/test_loaders.py]: Didn't find a way to test if `__init__` was called with `settings`

File: spider_feeder/parser.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import json
+from csv import DictReader
+
+from scrapy.exceptions import NotConfigured
+
+
+class TxtParser:
+
+    def __init__(self, settings):
```
Comment (ejulio) [inline spider_feeder/parser.py]: Not if would be better to declare the factory methods `from_settings` or `from_crawler` and then use `create_instance` from scrapy utils.

File: spider_feeder/loaders.py [hunk 3]
```diff
@@ -55,3 +65,12 @@ def _open(self, input_file_uri):
         open = load_object(self._file_handlers[parsed.scheme])
         logger.info(f'Opening file {input_file_uri} with scheme {parsed.scheme}.')
         return open(input_file_uri, encoding=self._file_encoding)
+
+    def _parse(self, fd, input_file_uri):
+        (_, file_extension) = path.splitext(input_file_uri)
+        file_extension = file_extension[1:]
+        parser_cls = load_object(self._file_parsers[file_extension])
+        print(self._file_parsers[file_extension])
```
Comment (raphapassini) [inline spider_feeder/loaders.py]: Logs are better than printing right? :P

Comment (ejulio) [reply spider_feeder/loaders.py]: :sweat_smile: thanks!

**Pull request merged**